### PR TITLE
Animate exploding messages on desktop

### PR DESCRIPTION
**Pull request by buoyad**
- Changes `ExplodingHeightRetainer` to throw away stored height on remount
  - Pipes `measure` through and uses `m.ordinal` instead of `m.id` to key the message

r? @keybase/react-hackers

### CODE REVIEW

**Comment by buoyad** [comment]
cc @cecileboucheron & @malgorithms if you'd like to change the colors, they're defined [here](https://github.com/keybase/client/pull/12317/files#diff-ecf15b17035331881283719eb6fdf71eR155). I chose arbitrarily from `globalColors`

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 1]
```diff
@@ -4,40 +4,73 @@ import * as ReactDOM from 'react-dom'
...
 import {urlsToImgSet} from '../../../../../common-adapters/icon.desktop'
 import {Box, ConnectedUsernames, Text} from '../../../../../common-adapters'
-import {collapseStyles, globalColors, styleSheetCreate} from '../../../../../styles'
+import {
+  collapseStyles,
+  glamorous,
+  globalColors,
+  platformStyles,
+  styleSheetCreate,
+} from '../../../../../styles'
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: so whats the native story here?

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 2]
```diff
@@ -4,40 +4,73 @@ import * as ReactDOM from 'react-dom'
...
+  React.Children.map(children, child => (child ? React.cloneElement(child) : child))
+
+const animationDuration = 1.5
+
+const retainedHeights = {}
 
 type State = {
-  height: ?number,
+  animating: boolean,
+  children: ?React.Node,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: why do we need to clone the children?

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 3]
```diff
@@ -4,40 +4,73 @@ import * as ReactDOM from 'react-dom'
...
     }
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: Props) {
     if (this.props.retainHeight) {
+      if (!prevProps.retainHeight && !this.timeoutID) {
+        // destroy local copy of children when animation finishes
+        this.setState({animating: true}, () => {
+          this.timeoutID = setTimeout(
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: likely need to clear the timeout if this gets his if one exists still

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 1 above)
Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: Nothing on native yet, I'll look into it today

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 2 above)
Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: As soon as the message explodes the text is deleted from the store; cloning the children is so the text doesn't disappear while the animation is running

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 3 above)
Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: done :+1: thanks

**Pull request merged**